examples/EinsteinToolkit/EinsteinToolkit.c

```diff
@@ -526,8 +524,22 @@ void setup(const char* program_source1, const char* program_source2)
   clGetContextInfo(context, CL_CONTEXT_DEVICES,
                    ndevice_ids*sizeof(cl_device_id), device_ids, NULL);
   assert(ndevice_ids >= 1);
-  device_id = device_ids[0];
-  
+  main_device_id = device_ids[0];
+
+  if (use_subdev)
+    {
+      const cl_device_partition_property props[]
+          = { CL_DEVICE_PARTITION_EQUALLY, 2, 0 };
+      cl_device_id subdevs[128];
+      cl_uint retval;
+      int err
+          = clCreateSubDevices (main_device_id, props, 2, subdevs, &retval);
```

Comment:
How do you verify that the sub devices is working correctly, that is, it's restricting its execution to only certain CUs? I think that's not trivial to do automatically.

Reply:
Actually it is. Currently it's not done automatically, but measuring execution time for 1 CU vs 2 CUs should do the trick.

Reply:
yes, that's true